tests/manage/test_dynamic_pvc_accessmodes_with_reclaim_policies.py

```diff
@@ -0,0 +1,375 @@
+            'test', 'pvc'
+        )
+        pvc_data['metadata']['namespace'] = self.namespace
+        pvc_data['spec']['storageClassName'] = sc_obj.name
+        pvc_data['spec']['accessModes'] = ['ReadWriteOnce']
+        pvc_obj = PVC(**pvc_data)
+        pvc_obj.create()
+
+        # Create two pods using same PVC
+        logger.info(f"Creating two pods using same PVC {pvc_obj.name}")
+
+        # Create first pod
+        if interface_type == constants.CEPHBLOCKPOOL:
+            pod_data = templating.load_yaml_to_dict(constants.CSI_RBD_POD_YAML)
+        elif interface_type == constants.CEPHFILESYSTEM:
+            pod_data = templating.load_yaml_to_dict(
+                constants.CSI_CEPHFS_POD_YAML
+            )
+
+        pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+            'test', 'pod'
+        )
+        pod_data['metadata']['namespace'] = self.namespace
+        pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = (
+            pvc_obj.name
+        )
+        pod_obj = helpers.create_pod(**pod_data)
+
+        node_pod1 = pod_obj.get()['spec']['nodeName']
+
+        # Create second pod
+        # Try creating pod until it is on a different node than first pod
+        for retry in range(1, 6):
+            pod_data['metadata']['name'] = helpers.create_unique_resource_name(
+                'test', 'pod'
+            )
+            pod_obj2 = helpers.create_pod(wait=False, **pod_data)
+            node_pod2 = pod_obj2.get()['spec']['nodeName']
+            if node_pod1 != node_pod2:
+                break
+            logger.info(
+                f"Both pods are on same node. Deleting second pod and "
+                f"creating another pod. Retry count:{retry}"
+            )
+            pod_obj2.delete()
+            if retry == 5:
+                raise UnexpectedBehaviour(
+                    "Second pod is always created on same node as of first "
+                    "pod even after trying 5 times."
+                )
+
+        # Run IO on first pod
+        logger.info(f"Running IO on first pod {pod_obj.name}")
+        pod_obj.run_io('fs', '512M')
+        logging.info(f"Waiting for IO results from pod {pod_obj.name}")
+        fio_result = pod_obj.get_fio_results()
+        logging.info("IOPs after FIO:")
+        logging.info(
+            f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
+        )
+        logging.info(
+            f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
+        )
+
+        # Fetch usage details
+        mount_point = pod_obj.exec_cmd_on_pod(command="df -kh")
+        mount_point = mount_point.split()
+        usage = mount_point[mount_point.index('/var/lib/www/html') - 1]
+
+        # Verify that second pod is not in Running state
+        try:
+            assert not pod_obj2.ocp.wait_for_resource(
+                condition='Running', resource_name=pod_obj2.name,
```

Comment:
why not just using get once?

Reply:
@sidhant-agrawal there could be multiple reasons for a pod to be stuck in "ContainerCreating" state. Hence, wouldn't it be better to validate that the 2nd pod is in ContainerCreating state due to the failure to mount an already mounted rbd volume ? But problem is, the error message is seen only in "oc describe pod> and in oc get events or in csi logs. 
**oc get pods -o yaml** wont have the failure reason.


e.g. 
Multi-Attach error for volume "pvc-7c80564b-a17c-11e9-b736-06fad4d9b278" Volume is already used by pod(s) pod-captain-rwo1

Reply:
> why not just using get once?

Every pod will remain in ContainerCreating initially for some time, thus to ensure the pod shouldn't come into 'Running' state after some time checking for some time.

Reply:
Now checking for the failure string in the oc describe pod command.